admin/attributes_controller.php

```diff
@@ -247,45 +247,69 @@
           $attributes_default = (int)$_POST['attributes_default'];
           $attributes_discounted = (int)$_POST['attributes_discounted'];
           $attributes_price_base_included = (int)$_POST['attributes_price_base_included'];
-
           $attributes_price_onetime = (float)$_POST['attributes_price_onetime'];
-          $attributes_price_factor = (float)$_POST['attributes_price_factor'];
-          $attributes_price_factor_offset = (float)$_POST['attributes_price_factor_offset'];
-          $attributes_price_factor_onetime = (float)$_POST['attributes_price_factor_onetime'];
-          $attributes_price_factor_onetime_offset = (float)$_POST['attributes_price_factor_onetime_offset'];
-          $attributes_qty_prices = zen_db_prepare_input($_POST['attributes_qty_prices']);
-          $attributes_qty_prices_onetime = zen_db_prepare_input($_POST['attributes_qty_prices_onetime']);
 
-          $attributes_price_words = (float)$_POST['attributes_price_words'];
-          $attributes_price_words_free = (int)$_POST['attributes_price_words_free'];
-          $attributes_price_letters = (float)$_POST['attributes_price_letters'];
-          $attributes_price_letters_free = (int)$_POST['attributes_price_letters_free'];
+          if (ATTRIBUTES_ENABLED_PRICE_FACTOR === 'true') {
+              $attributes_price_factor = (float)$_POST['attributes_price_factor'];
+              $attributes_price_factor_offset = (float)$_POST['attributes_price_factor_offset'];
+              $attributes_price_factor_onetime = (float)$_POST['attributes_price_factor_onetime'];
+              $attributes_price_factor_onetime_offset = (float)$_POST['attributes_price_factor_onetime_offset'];
+          } else {
+              $attributes_price_factor = 0;
+              $attributes_price_factor_offset = 0;
+              $attributes_price_factor_onetime = 0;
+              $attributes_price_factor_onetime_offset = 0;
+          }
+
+          if (ATTRIBUTES_ENABLED_QTY_PRICES === 'true') {
+            $attributes_qty_prices = zen_db_prepare_input($_POST['attributes_qty_prices']);
+            $attributes_qty_prices_onetime = zen_db_prepare_input($_POST['attributes_qty_prices_onetime']);
+          } else {
+            $attributes_qty_prices = '';
+            $attributes_qty_prices_onetime = '';
+          }
+
+          if (ATTRIBUTES_ENABLED_TEXT_PRICES === 'true') {
+              $attributes_price_words = (float)$_POST['attributes_price_words'];
+              $attributes_price_words_free = (int)$_POST['attributes_price_words_free'];
+              $attributes_price_letters = (float)$_POST['attributes_price_letters'];
+              $attributes_price_letters_free = (int)$_POST['attributes_price_letters_free'];
+          } else {
+              $attributes_price_words = 0;
+              $attributes_price_words_free = 0;
+              $attributes_price_letters = 0;
+              $attributes_price_letters_free = 0;
+          }
+
```

Comment:
could not all of this code be simplified using the coalesce `??` operator?

Reply:
Probably, but that would have required changes to each line; it was more expedient to simply break out the condition.

Reply:
... as well as identifying the configuration settings that were introducing the 'conflict'.

Reply:
identifying the config settings are completely unnecessary.

Reply:
If you want to rework each line to use the formatting you've specified, go for it!